src/Articles.jsx

```diff
@@ -1,25 +1,46 @@
 import { getArticles } from './Api'
 import { useEffect, useState } from "react"
+import { Link } from 'react-router-dom';
+
 
 function Articles() {
-const [articles, setArticles] = useState([])
+const [articles, setArticles] = useState([]);
+const [isLoading, setIsLoading] = useState(true);
+const [error, setError] = useState(null)
 
 useEffect(() => {
     getArticles().then((articlesFromApi) => {
-    setArticles(articlesFromApi)
+    setArticles(articlesFromApi);
+    setIsLoading(false);
+    })
+    .catch((err) => {
+        setError(err);
+        setIsLoading(false)
     })
 
 }, [])
 
+if (isLoading) {
+    return <div>Won't be long, just loading...</div>
+}
```

Comment:
You could also implement a _loading state_ like this on your single article page to comfort users who are awaiting the fetch request to complete 💡